### PR TITLE
kvm: Fixes #2763 move post-renewal libvirt restart class suitably

### LINKED ISSUE #2763: Host is still unsecured when provisionCertificate api is called
##### SUMMARY

When provisionCertificate api is executed the host ends up in unsecured state, instead should be secured

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API, agent
~~~

##### CLOUDSTACK VERSION

~~~
4.12
~~~

##### CONFIGURATION

ca.framework.cert.automatic.renewal = true
ca.plugin.root.allow.expired.cert = true
ca.plugin.root.auth.strictness = false

##### OS / ENVIRONMENT

NA

##### STEPS TO REPRODUCE

~~~
execute provisionCertificate api with a random kvm host with/out recconect=true, the api call is successful but connection is unsecured. 
~~~

##### EXPECTED RESULTS

~~~
Connection is secured 
~~~

##### ACTUAL RESULTS

~~~
Connection is unsecured 

~~~

[agent_unsecured.txt](https://github.com/apache/cloudstack/files/2222393/agent_unsecured.txt)
[unsecured_connection.txt](https://github.com/apache/cloudstack/files/2222394/unsecured_connection.txt)

### PR DESCRIPTION
**Pull request by rhtyd**
This fixes #2763 by moving a post cert-renewal class for kvm
plugin/hypervisor to src/main/java. The regression is due to change
in file-system layout due to maven standard refactoring on master and
issue was not caught during forward-merging of a PR from 4.11 branch.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

## Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [ ] I have added tests to cover my changes.
- [x] All relevant new and existing integration tests have passed.
- [x] A full integration testsuite with all test that can run on my environment has passed.

### CODE REVIEW

File: test/integration/smoke/test_vm_life_cycle.py [hunk 1]
```diff
@@ -918,7 +918,7 @@ def secure_all_hosts(self):
             cmd = provisionCertificate.provisionCertificateCmd()
             cmd.hostid = host.id
             cmd.reconnect = True
-            self.apiclient.updateConfiguration(cmd)
+            self.apiclient.provisionCertificate(cmd)
```
Comment (borisstoyanov) [inline test/integration/smoke/test_vm_life_cycle.py]: this one is already addressed in #2754 @rhtyd

Comment (rhtyd) [reply test/integration/smoke/test_vm_life_cycle.py]: okay

**Comment by rhtyd** [comment]
```
Test secure VM migration ... === TestName: test_01_secure_vm_migration | Status : SUCCESS ===
ok
Test Non-secured VM Migration ... === TestName: test_02_unsecure_vm_migration | Status : SUCCESS ===
ok
Test destroy Virtual Machine ... === TestName: test_03_secured_to_nonsecured_vm_migration | Status : SUCCESS ===
ok
Test Non-secured VM Migration ... === TestName: test_04_nonsecured_to_secured_vm_migration | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 4 tests in 402.693s

OK

```

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by borisstoyanov** [comment]
LGMT, the VM lifecycle test failed to deploy a VM, which does not seem related

**Pull request merged**